core/os/os.h

```diff
@@ -451,7 +450,7 @@ class OS {
 		SYSTEM_DIR_RINGTONES,
 	};
 
-	virtual String get_system_dir(SystemDir p_dir) const;
+	virtual String get_system_dir(SystemDir p_dir, bool p_shared_storage = true) const;
```

Comment:
@godotengine/android @akien-mga I'm looking for feedback on this approach to easily provide support for shared and app specific storage.
For most platform, this argument doesn't make a difference (hence default to `true`). On Android (and maybe iOS), we send different paths depending on whether this is `true` or `false`.

The alternatives to this approach were either:
- Create an app specific dir enum that mirrored the `SystemDir` enum
- Or add app specific dir values to the `SystemDir` enum

Reply:
I think the boolean option is fine for `3.x` (and can be forward ported to `master` too for consistency initially).

We might want to take time to rethink this API for 4.0 though, `get_system_dir()` is a very old method and it might not expose the best API to handle current features and requirements of modern OSes.

But it's also a fairly niche method that only a handful of users will need to call explicitly, which is why I think adding an Android-specific boolean flag in the short term is OK.